crates/cxx-qt-gen/src/writer/cpp/header.rs

```diff
@@ -130,7 +132,10 @@ pub fn write_cpp_header(generated: &GeneratedCppBlocks) -> String {
     .fold(generated.includes.clone(), |mut acc, qobject| {
```

Comment:
I find this whole iterator chain a bit confusing.

If I understand it correctly, the point is to create a BTreeSet of includes and then convert that to a list of strings.
However, the way it's written what's happening is this:
Iterate over the list of QObjects, create a BTreeSet of includes out of this.
Then discard that BTreeSet by converting it to an iterator again, then chain that iterator, fold it again into a new BTreeSet, just to then collect it again.

Could you maybe rewrite it to something like this (pseudocode):
```rust
let mut include_set = BTreeSet::new();
include_set.extend(
    generated.qobjects.iter()
         .flat_map(|qobject|qobject.blocks.includes)
         .cloned()
         .chain(genereated.extern_cxx_qt.iter().flat_map(block::includes).cloned())
);
include_set.into_iter().collect::<Vec<String>>().join("\n");
```
That would move the semantics to:
Create a BTreeSet out of this list of includes.
Then convert it to a `Vec<String>`.

Which at least to me seems a lot clearer. Maybe that's just me though.